src/lib/retire-js/RetireJsEngine.ts

```diff
@@ -0,0 +1,79 @@
+import {Logger} from '@salesforce/core';
+import {Controller} from '../../Controller';
+import {Config} from '../util/Config';
+import {RuleEngine} from '../services/RuleEngine';
+import {Catalog, Rule, RuleGroup, RuleResult, RuleTarget} from '../../types';
+import {ENGINE} from '../../Constants';
+//import childProcess = require('child_process');
+//import path = require('path');
+
+// Unlike the other engines we use, RetireJS doesn't really have "rules" per se. So we sorta have to synthesize a
+// "catalog" out of RetireJS's normal behavior.
+const retireJsCatalog: Catalog = {
+	rules: [{
+		engine: ENGINE.RETIRE_JS.valueOf(),
+		sourcepackage: ENGINE.RETIRE_JS.valueOf(),
+		// Give this rule an informative name, specific enough that we're able to supplement it with other rules later.
+		name: ' insecure-bundled-dependencies',
+		description: 'Identify bundled libraries/modules with known vulnerabilities.',
+		categories: ['Insecure Dependencies'],
+		rulesets: [],
+		languages: ['javascript'],
+		defaultEnabled: true
+	}],
+	categories: [{
+		engine: ENGINE.RETIRE_JS.valueOf(),
+		name: 'Insecure Dependencies',
+		paths: []
+	}],
+	rulesets: []
+};
+
+export class RetireJsEngine implements RuleEngine {
+	public static ENGINE_ENUM: ENGINE = ENGINE.RETIRE_JS;
+	public static ENGINE_NAME: string = ENGINE.RETIRE_JS.valueOf();
+	// RetireJS isn't really built to be invoked programmatically, so we'll need to invoke it as a CLI command. However, we
+	// can't assume that they have the module installed globally. So what we're doing here is identifying the path to the
+	// locally-scoped `retire` module, and then using that to derive a path to the CLI-executable JS script.
```

Comment:
Am I correct in assuming that the locally-scoped `retire` module will be from within our package? Also, I can't find this logic in the code - can I assume that this logic will be included as a part of `scanner:run` changes?

Reply:
Correct. The logic at line 38 is identifying the path to the local RetireJS module. It's commented out so the tests pass, and actually implemented in the `run` PR.